common/main/java/com/couchbase/lite/internal/fleece/MValue.java

```diff
@@ -46,6 +44,8 @@
  * <p>
  * The regrettable upside-down dependency on MValueConverter provides access to package
  * visible symbols in com.couchbase.lite.
+ * <p>
+ * It worries me that this isn't thread safe... but, as I say, I've never seen it be a problem.
```

Comment:
The synchronization happens in MutableArray and MutableDictionary so for most part, I think it would be OK. One skeptical part which is when setting mutate to the parent object may need to look through if a synchronization is needed there.

Reply:
Ah... I see.
I think this is might possibly be the mistaken belief that you only need to synchronize writes.  That's not how Java works: "If multiple threads access the same mutable state variable without synchronization, your program is broken." -- Brian Goetz